core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,372 @@
+        Preconditions.checkArgument(
+            defaultValue instanceof Boolean, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeBoolean((Boolean) defaultValue);
+        break;
+      case INTEGER:
+        Preconditions.checkArgument(
+            defaultValue instanceof Integer, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Integer) defaultValue);
+        break;
+      case LONG:
+        Preconditions.checkArgument(defaultValue instanceof Long, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Long) defaultValue);
+        break;
+      case FLOAT:
+        Preconditions.checkArgument(defaultValue instanceof Float, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Float) defaultValue);
+        break;
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue instanceof Double, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Double) defaultValue);
+        break;
+      case DATE:
+        Preconditions.checkArgument(
+            defaultValue instanceof Integer, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(DateTimeUtil.daysToIsoDate((Integer) defaultValue));
+        break;
+      case TIME:
+        Preconditions.checkArgument(
+            defaultValue instanceof Long, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(DateTimeUtil.microsToIsoTime((Long) defaultValue));
+        break;
+      case TIMESTAMP:
+        Preconditions.checkArgument(
+            defaultValue instanceof Long, "Invalid default %s value: %s", type, defaultValue);
+        if (((Types.TimestampType) type).shouldAdjustToUTC()) {
+          generator.writeString(DateTimeUtil.microsToIsoTimestamptz((Long) defaultValue));
+        } else {
+          generator.writeString(DateTimeUtil.microsToIsoTimestamp((Long) defaultValue));
+        }
+        break;
+      case STRING:
+        Preconditions.checkArgument(
+            defaultValue instanceof CharSequence, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(((CharSequence) defaultValue).toString());
+        break;
+      case UUID:
+        Preconditions.checkArgument(defaultValue instanceof UUID, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(defaultValue.toString());
+        break;
+      case FIXED:
+        Preconditions.checkArgument(
+            defaultValue instanceof ByteBuffer, "Invalid default %s value: %s", type, defaultValue);
+        ByteBuffer byteBufferValue = (ByteBuffer) defaultValue;
+        int expectedLength = ((Types.FixedType) type).length();
+        Preconditions.checkArgument(byteBufferValue.remaining() == expectedLength,
+            "Invalid default %s value: %s, incorrect length: %s",
+            type, defaultValue, byteBufferValue.remaining());
```

Comment:
`ByteBuffer#toString` isn't useful here. Just note that the size is wrong.

Reply:
Done.